accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -98,36 +97,33 @@ static class EpochState
 
             this.addedRanges = global.ranges.subtract(prevRanges).mergeTouching();
             this.removedRanges = prevRanges.mergeTouching().subtract(global.ranges);
-            this.prevSyncComplete = addedRanges.union(MERGE_ADJACENT, prevSyncComplete.subtract(removedRanges));
-            this.curSyncComplete = this.syncComplete = addedRanges;
+            this.syncComplete = addedRanges;
         }
 
-        boolean markPrevSynced(Ranges newPrevSyncComplete)
+        public boolean hasReachedQuorum()
         {
-            newPrevSyncComplete = newPrevSyncComplete.union(MERGE_ADJACENT, addedRanges).subtract(removedRanges);
-            if (prevSyncComplete.containsAll(newPrevSyncComplete))
-                return false;
-            Invariants.checkState(newPrevSyncComplete.containsAll(prevSyncComplete), "Expected %s to contain all ranges in %s; but did not", newPrevSyncComplete, prevSyncComplete);
-            prevSyncComplete = newPrevSyncComplete;
-            syncComplete = curSyncComplete.slice(newPrevSyncComplete, Minimal).union(MERGE_ADJACENT, addedRanges);
-            return true;
+            return syncTracker == null ? true : syncTracker.hasReachedQuorum();
         }
 
-        public boolean hasReachedQuorum()
+        private boolean recordSyncCompleteFromFuture()
```

Comment:
I actually want to rework this so its node based and not complete based... for testing it was easier to do complete based, but in practice node base will be best as the property is relative to the node, so node sending for epoch=N basically just sent for <= N as well